### DISCUSSION TITLE
EBUSD Adapter 5.0 HTTP Web Request Post

### DISCUSSION THREAD

**Discussion by physman38** [john30/ebusd#1097] [General]
Hello everyone,
hello @john30 

may I kindly ask a help here?

I have a Vaillant System and the Adapter 5.0, which is running fine.
I have a knx system as well with a Gira X1 and a RasPi - later with the EBUSD running.

The EBUS delivers HTTP site like eg.
`http://localhost:8088/data/470/hc1OPMode`
with response

```
{
 "470": {
  "messages": {   "Hc1OPMode": {
    "name": "Hc1OPMode",
    "passive": false,
    "write": false,
    "lastup": 1702890606,
    "zz": 21,
    "fields": {
     "0": {"name": "", "value": "auto"}
    }
   }
  }
 },
 "global": {
  "version": "23.2.23.2",
  "updatecheck": "OK, device firmware 1[3c11] available",
  "signal": true,
  "symbolrate": 22,
  "maxsymbolrate": 92,
  "minarbitrationmicros": 2,
  "maxarbitrationmicros": 152,
  "minsymbollatency": 6,
  "maxsymbollatency": 44,
  "qq": 49,
  "reconnects": 0,
  "masters": 3,
  "messages": 459,
  "lastup": 1702890606
 }
}

```

With the XML/JSON parser on the X1 I can get the values read out.
Via the RasPi I can read values via the 

ebusctl read Hc1OPMode

or i can write as well

ebusctl write -c 470 Hc1OPMode "2"

My question:
can I somehow via an HTTP Web Request POST set the Hc1OPMode?
What would be the code?

Thanks for you help!

**Comment by john30** [comment]
sure, see the wiki: https://github.com/john30/ebusd/wiki/3.2.-HTTP-client
or the openapi file: https://github.com/john30/ebusd/blob/master/contrib/html/openapi.yaml

**Comment by physman38** [comment]
Thanks @john30 !
However, I have still 2 question and ask kindly for advise:

1) for the HTTP readings and when I have done the Logik in the X1, I poll in X1 every 10sec, however the EBUSD does not update. Seems he is not sending data. Is my config maybe wrong?

EBUSD_OPTS="--configpath=/etc/ebusd/ebusd-configuration/ebusd-2.1.x/de --scanconfig=full -d enh:xxx.xxx.xxx.xxxx:9999 --pollinterval=30 --accesslevel=* --mqtthost=xxx.x.x.x --mqttport=8999 --mqttuser=USER --mqttpass=PASSWORD --httpport=8088 --mqttjson"

I do use as http line in X1 with the WebRequest Get method:
http://xxx.xxx.xxx.xxx:8088/data?required&poll=2&write

2) can you give me an easy example for writing to the EBUSD via http post?

many many thanks in advance

**Comment by felix-schwarz** [comment]
Did you find out how to write a value via HTTP @physman38? 

I'm currently struggling to replace
```
ebusctl write -c 700 HwcSFMode load
```
with a HTTP request.

Any help would be most welcome.

**Reply by physman38** [reply]
Unfortunately not..

**Reply by felix-schwarz** [reply]
Thanks for the reply. Are you using any alternative then? Like MQTT or directly running the `ebusctl` command?

**Reply by physman38** [reply]
I'm using MQTT for writing and HTTP for reading, as I have a GIRA X1, but the MQTT logic does seem to have some issues for reading I think. 
So I'm as well interested to know how one can set anything via HTTP write....

**Comment by john30** [comment]
a real value write is currently not part of the http endpoint, you can only issue write commands without input, e.g. one-time load of the buffer or such

**Comment by felix-schwarz** [comment]
@john30 Thanks for the answer! Are there any plans to allow writing commands with input through HTTP, too?
